Fabric.Terminology.SqlServer/Persistence/UnitOfWork/DtoDataReader.cs

```diff
@@ -0,0 +1,139 @@
+﻿namespace Fabric.Terminology.SqlServer.Persistence.UnitOfWork
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Data.Common;
+    using System.Linq;
+    using System.Reflection;
+
+    internal class DtoDataReader<T> : DbDataReader
+        where T : class
+    {
+        private readonly IReadOnlyCollection<T> dtos;
+
+        private readonly IEnumerator<T> enumerator;
+
+        private readonly List<PropertyInfo> properties = new List<PropertyInfo>();
+
+        private readonly Dictionary<string, int> nameLookup = new Dictionary<string, int>();
+
+        private DtoDataReader(IEnumerable<T> dtos)
```

Comment:
@j2jensen This works  but I really need to pass in a schema mapper between the dto and the actual database table.  As it is now, the dto **MUST** have properties listed in the same order as the columns in the database - e.g. it's really too brittle to leave long term and I'll add a TODO to come back and extend this class.  However, it's a great alternative to using a DataTable.